src/quickfix.c

```diff
@@ -951,6 +952,7 @@ typedef struct {
     char_u	*pattern;
     int		enr;
     int		type;
+    typval_T	*user_data;
```

Comment:
The support for garbage collection needs to be added for "user_data".  You can try setting a Dict as "user_data", invoke test_garbagecollect_now() function to do the garbage collection and then retrieve the user_data using getqflist() and try to access the Dict.

Reply:
Yeah that indeed seems to clear the dict. How do I add it to garbage collection though? `dict_alloc()`...?